plugins/woocommerce/src/Admin/Schedulers/SchedulerTraits.php

```diff
@@ -256,9 +243,14 @@ public static function do_action_or_reschedule() {
 		// or schedule to run now if no blocking jobs exist.
 		$blocking_job = static::get_next_blocking_job( $action_name );
 		if ( $blocking_job ) {
-			$after = new \DateTime();
+			$next_action_time = self::get_next_action_time( $blocking_job );
+
+			if ( ! is_a( $next_action_time, 'DateTime' ) || $next_action_time < new \DateTime() ) {
+				$next_action_time = new \DateTime();
+			}
```

Comment:
It would be appreciated if the relevant finding in #38600, why the value returned by `SchedulerTraits::get_next_action_time` may not be available for rescheduling and needs to be fallbacked, could be commented here.

Reply:
@eason9487 Comment added

Reply:
The comment refers to this:

https://github.com/woocommerce/action-scheduler/blob/cc041aa3cba238f1ad883e0ada833f0795ba3072/classes/ActionScheduler_ActionFactory.php#L304-L307

`ActionScheduler_SimpleSchedule` does not provide a next action time. And tbf, async/NullSchedule actions don't either.